Geometry_Engine/Query/Centroid.cs

```diff
@@ -204,48 +231,76 @@ public static Point Centroid(this PolyCurve curve, double tolerance = Tolerance.
             return new Point { X = xc, Y = yc, Z = zc };
 
         }
-        
+
         /***************************************************/
 
         public static Point Centroid(this Ellipse ellipse, double tolerance = Tolerance.Distance)
         {
             return ellipse.Centre;
         }
-                
+
         /***************************************************/
 
         public static Point Centroid(this Circle circle, double tolerance = Tolerance.Distance)
         {
             return circle.Centre;
         }
-        
+
         /***************************************************/
 
         public static Point Centroid(this Line line, double tolerance = Tolerance.Distance)
         {
             return line.PointAtParameter(0.5);
         }
-        
+
         /***************************************************/
 
         [NotImplemented]
         public static Point Centroid(this Arc arc, double tolerance = Tolerance.Distance)
         {
             throw new NotImplementedException();
         }
-        
+
         /***************************************************/
 
         [NotImplemented]
         public static Point Centroid(this NurbsCurve nurbsCurve, double tolerance = Tolerance.Distance)
         {
             throw new NotImplementedException();
         }
-        
+
         /***************************************************/
         /**** Private methods                           ****/
         /***************************************************/
 
+        private static Point Centroid(this IEnumerable<ICurve> outlines, IEnumerable<ICurve> openings = null, double tolerance = Tolerance.Distance)
```

Comment:
Given the comments above, feel we almost need to make this public, to be able to reach it from other projects.

Wonder if some system with including checks in here, and having a boolean input stating whether the checks should be preformed or not could be a solution... Not sure on this last part, but it will probably need to be made public.

Reply:
Agreed with @IsakNaslundBh, as in [my comment](https://github.com/BHoM/BHoM_Engine/issues/1705#issuecomment-671462380) I would even think of adding a description that this method requires closed and coplanar curves instead of any actual checks, moving the responsibility to the one that calls that method.